api/src/main/java/io/smallrye/stork/DefaultServiceInstance.java

```diff
@@ -8,21 +10,36 @@
 
     private final int port;
 
+    private final ServiceMetadata metadata;
+
     public DefaultServiceInstance(long id, String host, int port) {
+        this(id, host, port, new ServiceMetadata(Collections.emptyMap(), Collections.emptyMap()));
+    }
+
+    public DefaultServiceInstance(long id, String host, int port, ServiceMetadata metadata) {
         this.id = id;
         this.host = host;
         this.port = port;
+        this.metadata = metadata;
     }
 
+    @Override
     public long getId() {
         return id;
     }
 
+    @Override
     public String getHost() {
         return host;
     }
 
+    @Override
     public int getPort() {
         return port;
     }
+
+    @Override
+    public ServiceMetadata getMetadata() {
```

Comment:
I would name it ServiceInstanceMetadata for clarity. As there is a `Service` class.